### PR TITLE
fix: setup owner as CommunityDescription signer after minting community owner token

### PR DESCRIPTION
**Pull request by mprakhov**
When we mint the owner token, clients will start to verify the signature of the CommunityDescription from the contract owner, so CommunityDescription must be signed by the owner privateKey and not by generated community private key

Closes #https://github.com/status-im/status-desktop/issues/12473

### CODE REVIEW

File: protocol/messenger_handler.go [hunk 1]
```diff
@@ -2296,6 +2296,11 @@ func (m *Messenger) handleChatMessage(state *ReceivedMessageState, forceSeen boo
 		if err != nil {
 			return err
 		}
+
+		if communityResponse == nil {
+			return nil
+		}
```
Comment (mprakhov) [inline protocol/messenger_handler.go:2302]: fixed the crash, `handleWrappedCommunityDescriptionMessage` can return nil, nil

File: protocol/communities/manager.go [hunk 2]
```diff
@@ -4701,6 +4701,13 @@ func (m *Manager) AddCommunityToken(token *community_token.CommunityToken) (*Com
 		if err != nil {
 			return nil, err
 		}
+
+		if token.PrivilegesLevel == community_token.OwnerLevel {
+			community, err = m.updatePrivateKeyAndControlNode(community, m.identity)
```
Comment (jrainville) [inline protocol/communities/manager.go]: FYI, @osmaczko updated this method, so you'll have a conflict (whoever merges first)

**Review by jrainville — APPROVED** [review]
Looks good. Just be careful with the possible conflict with Patryk's PR

**Review by endulab — APPROVED** [review]

File: protocol/communities/manager.go (hunk 2 above)
Comment (osmaczko) [reply protocol/communities/manager.go]: The method is named `PromoteSelfToControlNode` now, there is no need to pass `m.identity`.

**Review by osmaczko — APPROVED** [review]

**Comment by mprakhov** [comment]
https://github.com/status-im/status-go/pull/4168#discussion_r1364517504
Is this done in some other PR? I do not observe such a fn name

P.S. Found, rebased, adapted to my needs

File: protocol/communities/manager.go [hunk 3]
```diff
@@ -4701,6 +4701,10 @@ func (m *Manager) AddCommunityToken(token *community_token.CommunityToken) (*Com
 		if err != nil {
 			return nil, err
 		}
+
+		if token.PrivilegesLevel == community_token.OwnerLevel {
+			m.promoteSelfToControlNode(community)
+		}
```
Comment (osmaczko) [inline protocol/communities/manager.go:4707]: @mprakhov I am analyzing the code in that area now, and after deeper thought, I'm struggling to understand why this change was necessary. Shouldn't only the control node be able to mint the owner token? This means the node has to already be a control node at that stage and `m.promoteSelfToControlNode(community)` is redundant :thinking:

Comment (mprakhov) [reply protocol/communities/manager.go:4707]: Due to the signature of the CommunityDescription. When we create the community, the community private key is different from the control node.
So when we deploy the contract owner (receiving owner token) - we change the community private key to be a private key of the contract owner. Otherwise, when you extract the signer through `unwrapCommunityDescriptionMessage` it will be signed by the community private key and not the owner private key and you'll not pass `HandleCommunityDescriptionMessage` due to `ErrNotAuthorized` error

Comment (mprakhov) [reply protocol/communities/manager.go:4707]: So yes, only the control node will mint the owner token, but as soon as we mint the owner token, all CommunityDescriptions verifications on the client side will be done through the loop, and loop check chat contact owner and signer of the CommunityDescription is always the same

Comment (osmaczko) [reply protocol/communities/manager.go:4707]: Right, you're correct. Sorry, I had a brain fart :sweat_smile:
